scim-core/src/test/java/org/apache/directory/scim/core/repository/PatchHandlerTest.java

```diff
@@ -56,6 +57,43 @@ public PatchHandlerTest() {
     this.patchHandler = new DefaultPatchHandler(schemaRegistry);
   }
 
+  @Test
+  public void applyReplaceEntireEnterpriseExtension() {
+    String enterpriseExtensionUrn = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User";
+    Map<String, String> enterpriseExtensionValue = new HashMap<>();
+    enterpriseExtensionValue.put("costCenter", "New Cost Center");
+    enterpriseExtensionValue.put("department", "New Department");
+    PatchOperation op = patchOperation(REPLACE, enterpriseExtensionUrn, enterpriseExtensionValue);
+    ScimUser updatedUser = patchHandler.apply(user(), List.of(op));
+    EnterpriseExtension actual = (EnterpriseExtension) updatedUser.getExtension("urn:ietf:params:scim:schemas:extension:enterprise:2.0:User");
```

Comment:
XS Nit:

This one is entirely my fault as I did this when I wrote the test, but we could replace the string here with the variable defined above
```suggestion
    EnterpriseExtension actual = (EnterpriseExtension) updatedUser.getExtension(enterpriseExtensionUrn);
```

Reply:
Pushed this cleanup to the branch